tests/test_feature_panel.py

```diff
@@ -0,0 +1,248 @@
+
+
+def test_is_feature_exist_in_disabled(session, feature, association_off):
+    feat = session.query(Feature).filter_by(name=feature.name).first()
+    assert internal.is_feature_exists_in_disabled(feat, session)
+
+
+def test_is_feature_enabled(mocker, session, association_on):
+    mocker.patch(
+        'app.internal.features.SessionLocal',
+        return_value=session
+    )
+    assert internal.is_feature_enabled(route='/route') is True
+
+
+def test_create_feature(session):
+
+    feat = internal.create_feature(
+        name='test1', route='/route', description='testing', creator='test'
+    )
+
+    assert feat.name == 'test1'
+
+
+def test_index(mocker, features_test_client, mock_features):
+    url = route.router.url_path_for('index')
+
+    resp = features_test_client.get(url)
+    assert resp.ok
+
+
+def test_add_feature_to_user(features_test_client, feature, form_mock):
+    url = route.router.url_path_for('add_feature_to_user')
+
+    resp = features_test_client.post(url, data=form_mock)
+    assert resp.ok
+
+
+def test_delete_user_feature_association(
+    features_test_client, form_mock, association_on
+):
+    url = route.router.url_path_for('delete_user_feature_association')
+
+    resp = features_test_client.post(url, data=form_mock)
+    assert resp.ok
+    assert resp.content == b'true'
+
+
+def test_enable_feature(features_test_client, form_mock, association_off):
+    url = route.router.url_path_for('enable_feature')
+
+    resp = features_test_client.post(url, data=form_mock)
+    assert resp.ok
+    assert resp.content == b'true'
+
+
+def test_disable_feature(features_test_client, form_mock, association_off):
+    url = route.router.url_path_for('disable_feature')
+
+    resp = features_test_client.post(url, data=form_mock)
+    assert resp.ok
+    assert resp.content == b'true'
+
+
+def test_show_user_enabled_features(mocker, features_test_client):
+
+    mocker.patch(
+        'app.routers.features.get_user_enabled_features',
+        return_value=True
+    )
+
+    url = route.router.url_path_for('show_user_enabled_features')
+
+    resp = features_test_client.get(url)
+    assert resp.ok
+    assert resp.content == b'true'
+
+
+def test_show_user_disabled_features(mocker, features_test_client):
+
+    mocker.patch(
+        'app.routers.features.get_user_disabled_features',
+        return_value=True
+    )
+
+    url = route.router.url_path_for('show_user_disabled_features')
+
+    resp = features_test_client.get(url)
+    assert resp.ok
+    assert resp.content == b'true'
+
+
+def test_get_user_unlinked_features(mocker, features_test_client, feature):
+    mocker.patch(
+        'app.routers.features.get_user_disabled_features',
+        return_value=True
+    )
+
+    url = route.router.url_path_for('get_user_unlinked_features')
+
+    resp = features_test_client.get(url)
+    assert resp.ok
+    json_resp = resp.json()
+    assert type(json_resp) is list
```

Comment:
Is it interesting to check the list len?

Reply:
What are you asking?

Reply:
Please if you have questions about the features panel ask them in the features panel PR, this is a icons PR.

Reply:
Do you want to validate len as well here?

Reply:
Did it you’ll see it in the next commit.